app/src/main/java/org/akvo/flow/util/SurveyFileNameGenerator.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright (C) 2017 Stichting Akvo (Akvo Foundation)
+ *
+ * This file is part of Akvo Flow.
+ *
+ * Akvo Flow is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Akvo Flow is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Akvo Flow.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package org.akvo.flow.util;
+
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+
+public class SurveyFileNameGenerator {
+
+    public SurveyFileNameGenerator() {
+    }
+
+    @NonNull
+    public String generateFileName(@Nullable String entryName) {
+        if (entryName == null || entryName.isEmpty()) {
```

Comment:
I would suggest we add the conditions of this block at the beginning of the `processFile()` method where we have other checks e.g. `entry.isDirectory()` otherwise it would be undesirable behaviour to have a survey in the table with an empty string as the name.

And in that case specify `entryName` as non-nullable?